compiler/src/iree/compiler/ConstEval/test/jit_globals.mlir

```diff
@@ -142,14 +141,13 @@ module @eval_i1_tensor {
 }
 
 // -----
-// CHECK-LABEL: @eval_i4_tensor
+// expected-error @+2 {{OUT_OF_RANGE; attempted to access an address outside of the valid buffer range}}
 module @eval_i4_tensor {
   util.global private @hoisted : tensor<5x6xi4>
   util.func public @main() -> tensor<5x6xi4> {
     %hoisted = util.global.load @hoisted : tensor<5x6xi4>
```

Comment:
https://github.com/openxla/iree/actions/runs/8727055167/job/23943706482?pr=17075#step:4:12961 lol ASan doesn't like this test case either

```
==62069==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x511000191297 at pc 0x55622c1d52a4 bp 0x7ffe370e7c90 sp 0x7ffe370e7450
  WRITE of size 1 at 0x511000191297 thread T0
      #0 0x55622c1d52a3 in __asan_memcpy (/work/build-asan/tools/iree-opt+0xc42a3) (BuildId: 5c7c7a74b0ce49f09aceaca7b0b926e0f25945ff)
      #1 0x7f9fe253e78d in mlir::iree_compiler::IREE::Util::raw_inplace_ostream::write_impl(char const*, unsigned long) /work/compiler/src/iree/compiler/Dialect/Util/IR/UtilAttrs.cpp:58:5
      #2 0x7f9fd9a75340 in llvm::raw_ostream::write(char const*, unsigned long) /work/third_party/llvm-project/llvm/lib/Support/raw_ostream.cpp:252:9
      #3 0x7f9fe25494e0 in mlir::iree_compiler::IREE::Util::serializeSplatValue(mlir::Location, mlir::Attribute, long, llvm::endianness, llvm::raw_ostream&) /work/compiler/src/iree/compiler/Dialect/Util/IR/UtilAttrs.cpp:209:8
      #4 0x7f9fe25494e0 in mlir::iree_compiler::IREE::Util::SerializableDenseElementsAttrModel::serializeToStream(mlir::Attribute, mlir::Location, llvm::endianness, llvm::raw_ostream&) const /work/compiler/src/iree/compiler/Dialect/Util/IR/UtilAttrs.cpp:795:14
      #5 0x7f9fe2548883 in mlir::iree_compiler::IREE::Util::SerializableDenseElementsAttrModel::serializeToBuffer(mlir::Attribute, mlir::Location, llvm::endianness, llvm::ArrayRef<char>) const /work/compiler/src/iree/compiler/Dialect/Util/IR/UtilAttrs.cpp:782:12
      #6 0x7f9fe2548883 in mlir::iree_compiler::IREE::Util::detail::SerializableAttrInterfaceInterfaceTraits::FallbackModel<mlir::iree_compiler::IREE::Util::SerializableDenseElementsAttrModel>::serializeToBuffer(mlir::iree_compiler::IREE::Util::detail::SerializableAttrInterfaceInterfaceTraits::Concept const*, mlir::Attribute, mlir::Location, llvm::endianness, llvm::ArrayRef<char>) /work/build-asan/compiler/src/iree/compiler/Dialect/Util/IR/UtilAttrInterfaces.h.inc:279:51
      #7 0x7f9fdca0055d in mlir::iree_compiler::ConstEval::FunctionCall::importSerializableAttr(mlir::Location, mlir::iree_compiler::IREE::Util::SerializableAttrInterface) /work/compiler/src/iree/compiler/ConstEval/Runtime.cpp:189:47
      #8 0x7f9fdca011b0 in mlir::iree_compiler::ConstEval::FunctionCall::addBufferViewArgumentAttr(mlir::Location, mlir::ShapedType, mlir::iree_compiler::IREE::Util::SerializableAttrInterface) /work/compiler/src/iree/compiler/ConstEval/Runtime.cpp:227:17
      #9 0x7f9fdca016b0 in mlir::iree_compiler::ConstEval::FunctionCall::addArgument(mlir::Location, mlir::Attribute) /work/compiler/src/iree/compiler/ConstEval/Runtime.cpp
```

Reply:
Just reverted the changes from https://github.com/openxla/iree/pull/16321 - looks broken. Should I also revert the test cases that PR added while I'm at it?